providers/keap/parse.go

```diff
@@ -15,3 +17,70 @@ func makeNextRecordsURL(moduleID common.ModuleID) common.NextPageFunc {
 		return jsonquery.New(node).StrWithDefault("next_page_token", "")
 	}
 }
+
+// Before parsing the records if applicable API will be called one time to get model describing custom fields.
+// Object will then be enhanced using model.
+func (c *Connector) parseReadRecords(
+	ctx context.Context, config common.ReadParams, jsonPath string,
+) common.RecordsFunc {
+	return func(node *ajson.Node) ([]map[string]any, error) {
+		arr, err := jsonquery.New(node).Array(jsonPath, true)
+		if err != nil {
+			return nil, err
+		}
+
+		customFields, err := c.requestCustomFields(ctx, config.ObjectName)
```

Comment:
If this process (of enhancing the custom fields) fails, then the whole read operation fails. Can you make this optional? i.e. If we are able to add in human-readable names to the data, great, else, we should just log it and move on instead of completely failing

Reply:
This one is interconnected with the ListObjectMetadata PR.

Few impressions/ideas I have at the moment:
* ListObjectMetadata returns fields that client uses to select :ballot_box_with_check: to configure reading objects. If unlucky the UI will display some fields. Assumption: This doesn't happen often.
* In case of incomplete success it would be misleading to only return custom_fields without inlined property. This will blur why particular custom field is missing, because of actual null or because of hidden error that we only know of. Assumption: clients rely on `ReadResultRow.Fields`.

Let me know, if my judgment is relevant. I had a feeling that it counts as an error during development. But now I am not sure.

Reply:
cc @laurenzlong